src/main/java/ru/itis/controllers/WishListController.java

```diff
@@ -28,28 +28,30 @@ public WishListController(ItemService itemService, WishListService wishListServi
     @GetMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Get a single wish list")
-    public WishListDto getWishList(@PathVariable Long listId) {
+    public WishListDto getWishList(@PathVariable("list-id") Long listId, @RequestHeader(name = "AUTH") String token) {
+//      TODO: if anonym --> ability to reserve an item
         Optional<WishList> wishListCandidate = wishListService.findWishListById(listId);
```

Comment:
It is very bad having Null Object and returning optional at the same time. You can just return default object.
```java
WishList wishListService.findWishListById(listId) {
    return repository.byId(listId).orElse(defaultWishList);
}
```

Reply:
fixed